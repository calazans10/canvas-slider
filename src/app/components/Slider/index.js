import PropTypes from 'prop-types';
import { Layer } from 'react-konva';
import Image from '../Image';
import { getNewPosition } from '../../../helpers';
import SliderStyle from './style';

function Slider({ width, height, images }) {
  const dragBoundaryLimit = width * (images.length - 1) * -1;

  return (
    <SliderStyle width={width} height={height}>
      <Layer draggable dragBoundFunc={(pos) => getNewPosition(pos, dragBoundaryLimit)}>
        {images.map((image, index) => {
          const position = width * index;

          return (
            <Image
              key={`${position}-${image}`}
              src={image}
              position={position}
              sliderWidth={width}
              sliderHeight={height}
            />
          );
        })}
      </Layer>
    </SliderStyle>
  );
}

Slider.propTypes = {
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  images: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Slider;
