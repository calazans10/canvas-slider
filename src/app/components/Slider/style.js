import { Stage } from 'react-konva';
import styled from 'styled-components';

const SliderStyle = styled(Stage)`
  border: 1px solid #ccc;
`;

export default SliderStyle;
