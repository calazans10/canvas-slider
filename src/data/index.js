import bob0 from '../images/0.jpg';
import bob1 from '../images/1.jpg';
import bob2 from '../images/2.jpg';
import bob3 from '../images/3.jpg';
import bob4 from '../images/4.jpg';
import bob5 from '../images/5.jpg';
import bob6 from '../images/6.jpg';

// eslint-disable-next-line import/prefer-default-export
export const images = [bob0, bob1, bob2, bob3, bob4, bob5, bob6];
