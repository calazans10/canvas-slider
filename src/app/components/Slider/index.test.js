import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { images } from '../../../data';
import Slider from '.';

describe('Slider', () => {
  test('renders without crashing', async () => {
    const wrapper = shallow(<Slider width={640} height={400} images={images} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
