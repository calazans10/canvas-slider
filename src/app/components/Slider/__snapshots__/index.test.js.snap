// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders without crashing 1`] = `
<Styled(Component)
  height={400}
  width={640}
>
  <Layer
    dragBoundFunc={[Function]}
    draggable={true}
  >
    <Image
      key="0-0.jpg"
      position={0}
      sliderHeight={400}
      sliderWidth={640}
      src="0.jpg"
    />
    <Image
      key="640-1.jpg"
      position={640}
      sliderHeight={400}
      sliderWidth={640}
      src="1.jpg"
    />
    <Image
      key="1280-2.jpg"
      position={1280}
      sliderHeight={400}
      sliderWidth={640}
      src="2.jpg"
    />
    <Image
      key="1920-3.jpg"
      position={1920}
      sliderHeight={400}
      sliderWidth={640}
      src="3.jpg"
    />
    <Image
      key="2560-4.jpg"
      position={2560}
      sliderHeight={400}
      sliderWidth={640}
      src="4.jpg"
    />
    <Image
      key="3200-5.jpg"
      position={3200}
      sliderHeight={400}
      sliderWidth={640}
      src="5.jpg"
    />
    <Image
      key="3840-6.jpg"
      position={3840}
      sliderHeight={400}
      sliderWidth={640}
      src="6.jpg"
    />
  </Layer>
</Styled(Component)>
`;
